doc/read-the-docs-site/adr/0008-marconi-sidechain-testing-strategy.rst

```diff
@@ -0,0 +1,96 @@
+  Reports produced for a release should be hosted indefinitely.
+
+UNIT TESTING:
+
+* We will test indexers and filters with as many positive and negative combinations as possible using data-driven and property tests.
+
+* We will not attempt to test the CLI or RPC interface at unit level because we already have some sanity tests being run on commit on private testnet for fast feedback.
+
+* We will plan for and produce tests alongside implementation of new product features.
+
+INTEGRATION TESTING:
+
+* We will plan test coverage and also aim to produce them alongside implementation of new product features.
+  It is acceptable to have separate tasks to produce the tests but them must be complete within the same PI cycle.
+
+* We will produce sanity tests to be to be executed on PR commit and run on a private testnet (using cardano-testnet) which will be primed with a basic dataset not intended for performance evaluations or querying in a complex environment.
+  These tests should only check the essential functionality of each feature.
+  See https://github.com/input-output-hk/marconi/issues/93 for checklist of test cases.
+
+END-TO-END TESTING:
+
+* We will plan test coverage and also aim to produce them alongside implementation of new product features.
+  It is acceptable to have separate tasks to produce the tests but them must be complete within the same PI cycle.
+
+* We will sometimes perform exploratory testing on major features, such as new RPC methods or cli options.
+  This can provide faster feedback in lieu of automation testing being produced.
+  Should not be carried out by the engineer who implemented the feature (due to assumption bias).
+    
+* We will produce tests for each of the RPC API methods to stress the filters
+  These tests will assert against known correct response content.
+  These tests will ideally be run on mainnet because there is more interesting transaction data there and so are more likely to catch an edge-case.
+  These tests should be run either after a sync test or on an instance of the process under test that is always running fully synced.
+  See RPC-1 in https://github.com/input-output-hk/marconi/issues/93 for checklist of test cases.
+
+* We will produce tests for the RPC API methods to cover a board set of response data comparing against a source of truth
+  These tests will compare against results produced by `cardano-db-sync <https://github.com/input-output-hk/cardano-db-sync>`_
+  These tests will ideally be run on mainnet
+  see RPC-2 in https://github.com/input-output-hk/marconi/issues/93 for checklist of test cases.
+
+* We will not have any end-to-end tests that query maroni's db because this does not always store relevant data and it does not resemble how the user should be querying.
+
+* We will produce sync performance tests to be run on each public network: preview, preprod and mainnet. (See ADR 7: Marconi observability for details)
```

Comment:
do we set threshold that would lead to failing tests result?

Reply:
Yeah, definitely. 
I didn't want to put details on performance testing here but it is worth referencing the other ADR for those looking.

Reply:
I think we need to mention this in this ADR. The other ADR is about identify how to "observe" the behaviour of `marconi-sidechain`.
Maybe put anything related to performance tests wrt to sync time, memory usage, query performance, etc should go here?

Reply:
Thanks for the feedback. I have now added some more detail on the sync and query performance testing.